File: cathode/src/main/java/net/simonvt/cathode/ui/lists/ListsAdapter.java
/*
 * Copyright (C) 2016 Simon Vig Therkildsen
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package net.simonvt.cathode.ui.lists;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import butterknife.BindView;
import butterknife.ButterKnife;
import net.simonvt.cathode.R;
import net.simonvt.cathode.common.entity.UserList;
import net.simonvt.cathode.common.ui.adapter.BaseAdapter;
import net.simonvt.cathode.provider.DatabaseContract.ListsColumns;

public class ListsAdapter extends BaseAdapter<UserList, ListsAdapter.ViewHolder> {

  public interface OnListClickListener {

    void onListClicked(long listId, String listName);
  }

  private OnListClickListener listener;

  public ListsAdapter(OnListClickListener listener, Context context) {
    super(context);
    this.listener = listener;
    setHasStableIds(true);
  }

  @Override public long getItemId(int position) {
    return getList().get(position).getId();
  }

  @Override
  protected boolean areItemsTheSame(@NonNull UserList oldItem, @NonNull UserList newItem) {
    return oldItem.getId() == newItem.getId();
  }

  @Override public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
    View v = LayoutInflater.from(getContext()).inflate(R.layout.list_row_list, parent, false);
    final ViewHolder vh = new ViewHolder(v);

    v.setOnClickListener(new View.OnClickListener() {
      @Override public void onClick(View v) {
        listener.onListClicked(vh.getItemId(), vh.name.getText().toString());
      }
    });

    return vh;
  }

  @Override public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
    UserList userList = getList().get(position);

    holder.name.setText(userList.getName());
    holder.description.setText(userList.getDescription());
  }

  public static class ViewHolder extends RecyclerView.ViewHolder {

    @BindView(R.id.name) TextView name;

    @BindView(R.id.description) TextView description;

    public ViewHolder(View itemView) {
      super(itemView);
      ButterKnife.bind(this, itemView);
    }
  }
}

File: cathode/src/main/java/net/simonvt/cathode/ui/lists/ListsAdapter.kt
/*
 * Copyright (C) 2016 Simon Vig Therkildsen
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package net.simonvt.cathode.ui.lists

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import net.simonvt.cathode.common.ui.adapter.BaseAdapter
import net.simonvt.cathode.databinding.ListRowListBinding
import net.simonvt.cathode.entity.UserList
import net.simonvt.cathode.ui.lists.ListsAdapter.ViewHolder

class ListsAdapter(private val listener: OnListClickListener, context: Context?) :
  BaseAdapter<UserList?, ViewHolder?>(context) {

  interface OnListClickListener {
    fun onListClicked(listId: Long, listName: String)
  }

  override fun getItemId(position: Int): Long {
    return list[position]!!.id
  }

  override fun areItemsTheSame(oldItem: UserList, newItem: UserList): Boolean {
    return oldItem.id == newItem.id
  }

  override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
    val binding = ListRowListBinding.inflate(LayoutInflater.from(context), parent, false)
    val holder = ViewHolder(binding)
    binding.root.setOnClickListener {
      listener.onListClicked(
        holder.itemId,
        holder.name.text.toString()
      )
    }
    return holder
  }

  override fun onBindViewHolder(holder: ViewHolder, position: Int) {
    val list = list[position]!!
    holder.name.text = list.name
    holder.description.text = list.description
  }

  class ViewHolder(binding: ListRowListBinding) : RecyclerView.ViewHolder(binding.root) {
    val name = binding.name
    val description = binding.description
  }

  init {
    setHasStableIds(true)
  }
}
